exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogRecordReader.java

```diff
@@ -0,0 +1,261 @@
+    try {
+      Path hdfsPath = new Path(inputPath);
+      Configuration conf = new Configuration();
+      FSDataInputStream fsStream = fileSystem.open(hdfsPath);
+      CompressionCodecFactory factory = new CompressionCodecFactory(conf);
+      CompressionCodec codec = factory.getCodec(hdfsPath);
+      if (codec == null) {
+        reader = new BufferedReader(new InputStreamReader(fsStream.getWrappedStream(), "UTF-8"));
+      } else {
+        CompressionInputStream comInputStream = codec.createInputStream(fsStream.getWrappedStream());
+        reader = new BufferedReader(new InputStreamReader(comInputStream));
+      }
+      this.inputPath = inputPath;
+      this.lineCount = 0;
+      this.config = config;
+      this.buffer = fragmentContext.getManagedBuffer(4096);
+      setColumns(columns);
+
+    } catch (IOException e) {
+      logger.debug("Log Reader Plugin: " + e.getMessage());
+    }
+  }
+
+  public void setup(final OperatorContext context, final OutputMutator output) throws ExecutionSetupException {
+    this.writer = new VectorContainerWriter(output);
+    String regex = config.getPattern();
+
+
+    fieldNames = config.getFieldNames();
+    dataTypes = config.getDataTypes();
+    dateFormat = config.getDateFormat();
+    timeFormat = config.getTimeFormat();
+    errorOnMismatch = config.getErrorOnMismatch();
+
+    /*
+    This section will check for;
+    1.  Empty regex
+    2.  Invalid Regex
+    3.  Empty date string if the date format is used
+    4.  No capturing groups in the regex
+    5.  Incorrect number of data types
+    6.  Invalid data types
+     */
+    if (regex.isEmpty()) {
+      throw UserException.parseError().message("Log parser requires a valid, non-empty regex in the plugin configuration").build(logger);
+    } else {
+      //TODO Check for invalid regex
+      r = Pattern.compile(regex);
+      Matcher m = r.matcher("test");
+      if (m.groupCount() == 0) {
+        throw UserException.parseError().message("Invalid Regular Expression: No Capturing Groups", 0).build(logger);
+      } else if (m.groupCount() != (fieldNames.size())) {
+        throw UserException.parseError().message("Invalid Regular Expression: Field names do not match capturing groups.  There are " + m.groupCount() + " captured groups in the data and " + fieldNames.size() + " specified in the configuration.", 0).build(logger);
+
+      } else if ((dataTypes == null) || m.groupCount() != dataTypes.size()) {
+        //If the number of data types is not correct, create a list of varchar
+        dataTypes = new ArrayList<String>();
+        for (int i = -0; i < m.groupCount(); i++) {
+          dataTypes.add("VARCHAR");
+        }
+      }
+    }
+
+    //Check and set up date formats
+    if (dataTypes.contains("DATE") || dataTypes.contains("TIMESTAMP")) {
+      if (dateFormat != null && !dateFormat.isEmpty()) {
+        df = new java.text.SimpleDateFormat(dateFormat);
```

Comment:
I believe this will throw an exception for a bad format. Should catch it.

Also, OK to use `SimpleDateFormat` and import the class. This code does not have the same draconian restrictions as UDFs.